packages/@lwc/integration-karma/test/component/LightningElement.errorCallback/index.spec.js

```diff
@@ -205,3 +222,174 @@ describe('error boundary during initial component construction', () => {
+        };
+
+        beforeEach(() => {
+            // Overriding window.onerror disables Jasmine's global error handler, so we can listen for errors
+            // ourselves. There doesn't seem to be a better way to disable Jasmine's behavior here.
+            // https://github.com/jasmine/jasmine/pull/1860
+            originalOnError = window.onerror;
+            // Dummy onError because Jasmine tries to call it in case of a rejection:
+            // https://github.com/jasmine/jasmine/blob/169a2a8/src/core/GlobalErrors.js#L104-L106
+            window.onerror = () => {};
+            caughtError = undefined;
+            window.addEventListener('error', onError);
+            window.addEventListener('unhandledrejection', onRejection);
+        });
+
+        afterEach(() => {
+            window.removeEventListener('error', onError);
+            window.removeEventListener('unhandledrejection', onRejection);
+            window.onerror = originalOnError;
+        });
+
+        function testStub(testcase, hostSelector, hostClass, expectAfterThrowingChildToExist) {
+            it(`parent errorCallback throws after value mutation ${testcase}`, () => {
+                const throwElm = createElement(hostSelector, { is: hostClass });
+                const noThrowElm = createElement('x-no-throw-on-mutate', { is: XNoThrowOnMutate });
+                document.body.appendChild(throwElm);
+                document.body.appendChild(noThrowElm);
+                return (
+                    Promise.resolve()
+                        .then(() => {
+                            throwElm.show = true;
+                            noThrowElm.show = true;
+                        })
+                        // Need to wait a few ticks so flushRehydrationQueue can finish
+                        .then(() => new Promise((resolve) => setTimeout(resolve)))
+                        .then(() => new Promise((resolve) => setTimeout(resolve)))
+                        .then(() => {
+                            // error is thrown by parent's errorCallback
+                            expect(caughtError).not.toBeUndefined();
+                            expect(caughtError.message).toMatch(
+                                /error in the parent error callback after value mutation/
+                            );
+                            // child after the throwing child is not rendered
+                            // TODO [#3261]: strange observable difference between native vs synthetic lifecycle
+                            const afterThrowingChild =
+                                throwElm.shadowRoot.querySelector('x-after-throwing-child');
+                            if (expectAfterThrowingChildToExist) {
+                                expect(afterThrowingChild).not.toBeNull();
+                            } else {
+                                expect(afterThrowingChild).toBeNull();
+                            }
```

Comment:
This observable difference is odd to me so I filed a bug: https://github.com/salesforce/lwc/issues/3261

Reply:
Do you think it's worth adding test cases for the observable differences?

Reply:
The observable difference is tested above by the `if/else` branch, yes. 🙂